### PR TITLE
Enable nullability in DataGridViewEditingControlShowingEventArgs

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DataGridViewEditingControlShowingEventArgs.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6135)

### CODE REVIEW

**Comment by RussKie** [comment]
image

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewEditingControlShowingEventArgs.cs [hunk 1]
```diff
@@ -2,21 +2,19 @@
...
-#nullable disable
-
 namespace System.Windows.Forms
 {
     public class DataGridViewEditingControlShowingEventArgs : EventArgs
     {
         private DataGridViewCellStyle _cellStyle;
 
-        public DataGridViewEditingControlShowingEventArgs(Control control, DataGridViewCellStyle cellStyle)
+        public DataGridViewEditingControlShowingEventArgs(Control? control, DataGridViewCellStyle cellStyle)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewEditingControlShowingEventArgs.cs:11]: I think the right thing to do here is to make the `control` not nullable as well.
image

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewEditingControlShowingEventArgsTests.cs [hunk 2]
```diff
@@ -11,31 +11,31 @@ public class DataGridViewEditingControlShowingEventArgsTests : IClassFixture<Thr
...
+            var e = new DataGridViewEditingControlShowingEventArgs(button, cellStyle);
+            Assert.Equal(button, e.Control);
+            Assert.Equal(cellStyle, e.CellStyle);
         }
 
         [Fact]
         public void CellStyle_SetNull_ThrowsArgumentNullException()
         {
-            var e = new DataGridViewEditingControlShowingEventArgs(null, null);
+            var e = new DataGridViewEditingControlShowingEventArgs(new Button(), new DataGridViewCellStyle());
```
Comment (dreddy-work) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewEditingControlShowingEventArgsTests.cs:38]:
```suggestion
           using var button = new Button();
            var e = new DataGridViewEditingControlShowingEventArgs(button, new DataGridViewCellStyle());
```

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Docs: https://github.com/dotnet/docs/issues/26720

**Pull request merged**